src/xma/src/xmaplugin/xmaplugin.cpp

```diff
@@ -375,12 +310,7 @@ xma_plg_buffer_write(XmaSession s_handle,
         return XMA_SUCCESS;
     }
 
-    rc = xclSyncBO(b_obj_priv->dev_handle, b_obj_priv->boHandle, XCL_BO_SYNC_BO_TO_DEVICE, size, offset);
-    if (rc != 0) {
-        xma_logmsg(XMA_ERROR_LOG, XMAPLUGIN_MOD, "xma_plg_buffer_write failed. dev_index: %d. xclSyncBO failed. Check device logs for more info. Error: %d", b_obj_priv->dev_index, rc);
-        return XMA_ERROR;
-    }
-
+    b_obj_priv->xrt_bo.sync(XCL_BO_SYNC_BO_TO_DEVICE, size, offset);
```

Comment:
@stsoe , currently we are relying on system error thrown by xrt::bo::sync, do we need to explicitly handle this and throw XMA_ERROR?

Reply:
added try catch block and returning XMA_ERROR when exception caught.